### ISSUE TITLE
Add contextual options to toggle alignment/snapping properties

### ISSUE THREAD

**Issue by avvvvve** [musescore/MuseScore#23329]
### Context
4.4 introduces new alignment properties for these elements:

- **Hairpins**, which have both "Snap to previous" and "Snap to next" toggles
     <img width="303" alt="image" src="https://github.com/musescore/MuseScore/assets/20806406/398b14c4-7990-4c23-8765-3612afbabe53">

- **Tempo lines**, which just has "Snap to next"
     <img width="303" alt="image" src="https://github.com/musescore/MuseScore/assets/20806406/64dbe7d5-acad-4be8-9f8f-aaa096f06a3f">

### Requirement 1
Create two new shortcut-assignable actions in Preferences > Shortcuts:
- Toggle snap to previous
- Toggle snap to next

They don't need a default shortcut. These will toggle the properties for both hairpins and tempo lines (for the latter, only "Toggle snap to next").

### Requirement 2
Add context menu actions when hairpins and tempo text are right-clicked to control the snapping toggles in properties. The option(s) should be added after "Edit element" and before "Style", with divider lines above/below.

Hairpins:
<img width="434" alt="image" src="https://github.com/musescore/MuseScore/assets/20806406/102953db-8429-4fc5-9b2f-68f8a780d287">

Tempo text:
<img width="385" alt="image" src="https://github.com/musescore/MuseScore/assets/20806406/4baeba24-116b-4847-bc83-7a4825c6adc5">
